CosmosGettingStartedDotnetCoreTutorial/Program.cs

```diff
@@ -118,21 +118,8 @@ private async Task AddItemsToContainerAsync()
                 IsRegistered = false
             };
 
-            // Read the item to see if it exists. Note ReadItemAsync will not throw an exception if an item does not exist. Instead, we check the StatusCode property off the response object. 
-            ItemResponse<Family> andersenFamilyResponse = await this.container.ReadItemAsync<Family>(andersenFamily.Id, new PartitionKey(andersenFamily.LastName));
-
-            if (andersenFamilyResponse.StatusCode == HttpStatusCode.NotFound)
-            {
-                // Create an item in the container representing the Andersen family. Note we provide the value of the partition key for this item, which is "Andersen"
-                andersenFamilyResponse = await this.container.CreateItemAsync<Family>(andersenFamily, new PartitionKey(andersenFamily.LastName));
-
-                // Note that after creating the item, we can access the body of the item with the Resource property off the ItemResponse. We can also access the RequestCharge property to see the amount of RUs consumed on this request.
-                Console.WriteLine("Created item in database with id: {0} Operation consumed {1} RUs.\n", andersenFamilyResponse.Resource.Id, andersenFamilyResponse.RequestCharge);
-            }
-            else
-            {
-                Console.WriteLine("Item in database with id: {0} already exists\n", andersenFamilyResponse.Resource.Id);
-            }
+            // Upsert the item, i.e. if it exists it gets updated, otherwise it gets created
+            ItemResponse<Family> andersenFamilyResponse = await this.container.UpsertItemAsync<Family>(andersenFamily, new PartitionKey(andersenFamily.LastName));

```

Comment:
I believe this beats the purpose of the original code, while I agree that Upsert solves the exception, the code was showing how to handle a scenario where an item might Not exist.

Possible try/catch CosmosException when StatusCode == NotFound

Reply:
@ealsur Thanks for the feedback. I added try catch semantics to the code. Keep in mind that the method Container.ReadItemAsync<T> returns an exception when the code runs. Therefore, since the Cosmos Database and its container are empty the catch block will always execute and the items will get created in each catch clause.